### PR TITLE
use dictionary to deal with multiple dice models

### PR DESCRIPTION
**Pull request by ppoliani**
This was done so we can have quick lookups into the individual dice model and thus make the model updates in a cleaner way.

### CODE REVIEW

File: Main-Dynamic-Child-Components.elm [hunk 1]
```diff
@@ -1,6 +1,7 @@
 module Main exposing (..)
 
-import Dice exposing (init)
+import Dict exposing (..)
+import Dice exposing (init, initialModel)
```
Comment (slyg) [inline Main-Dynamic-Child-Components.elm:4]: I do not see the Dice module expose `initialModel`, maybe we can simply use Dice.init in this case

File: Main-Dynamic-Child-Components.elm [hunk 2]
```diff
@@ -25,7 +26,7 @@ type alias DiceId =
 
 
 type alias Model =
-    { diceList : List ( DiceId, Dice.Model )
+    { diceDict : Dict DiceId Dice.Model
```
Comment (slyg) [inline Main-Dynamic-Child-Components.elm:29]: 👍

File: Main-Dynamic-Child-Components.elm (hunk 1 above)
Comment (ppoliani) [reply Main-Dynamic-Child-Components.elm:4]: Oh shit. True. I accepted your changes when fixing the conflict :(

Comment (slyg) [reply Main-Dynamic-Child-Components.elm:4]: 😬

File: Main-Dynamic-Child-Components.elm [hunk 3]
```diff
@@ -49,73 +48,70 @@ type Msg
...
     | Add
     | Remove DiceId
-    | SubMsg DiceId Dice.Msg
+    | DiceMsg DiceId Dice.Msg
 
 
-initDice : ( DiceId, Dice.Model ) -> ( DiceId, Dice.Model )
-initDice ( id, _ ) =
+initDice : a -> b -> Dice.Model
+initDice _ _ =
```
Comment (slyg) [inline Main-Dynamic-Child-Components.elm:55]: I assume this signature is due to `Dict.map`

Comment (ppoliani) [reply Main-Dynamic-Child-Components.elm:55]: yes exactly

**Review by slyg — APPROVED** [review]
Thx dude !

**Pull request merged**